motor-control-module/src/main/java/org/sagebionetworks/research/motor_control_module/step_view/MPowerActiveUIStepView.java

```diff
@@ -0,0 +1,49 @@
+package org.sagebionetworks.research.motor_control_module.step_view;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.sagebionetworks.research.domain.step.interfaces.Step;
+import org.sagebionetworks.research.motor_control_module.step.AppStepType;
+import org.sagebionetworks.research.motor_control_module.step.MPowerActiveUIStep;
+import org.sagebionetworks.research.presentation.DisplayString;
+import org.sagebionetworks.research.presentation.mapper.DrawableMapper;
+import org.sagebionetworks.research.presentation.model.ColorThemeView;
+import org.sagebionetworks.research.presentation.model.ImageThemeView;
+import org.sagebionetworks.research.presentation.model.action.ActionView;
+import org.sagebionetworks.research.presentation.model.implementations.ActiveUIStepViewBase;
+import org.sagebionetworks.research.presentation.model.interfaces.ActiveUIStepView;
+import org.threeten.bp.Duration;
+
+public class MPowerActiveUIStepView extends ActiveUIStepViewBase {
+    public static final String TYPE = AppStepType.MPOWER_ACTIVE;
+
+    public MPowerActiveUIStepView(@NonNull String identifier, int navDirection, @NonNull ImmutableMap<String, ActionView> actions,
```

Comment:
What's going on with these epic constructors?  Is it something the builder pattern should be used for?

Reply:
I also recommend new POJOs using @AutoValue for the constructor issue and for the auto-generated equals, hashCode, toString, Parcelable, and GsonAdapter. 

We have a mix of handwritten classes for existing POJOs